### DISCUSSION TITLE
Eclipse keyboard input problem

### DISCUSSION THREAD

**Discussion by Alex-Farol** [netblue30/firejail#4718] [Q&A] [answered]
Hello!

I’m trying to run Eclipse 2021.09 with firejail on a Debian Bullseye (version 0.9.64.4), but it can’t get keyboard input. Launching it from terminal, I get messages like “IBUS-WARNING **: 08:50:29.000: Unable to connect to ibus”. Outside jail, it works as expected.

Thanks in advance.

**Comment by rusty-snake** [comment, marked as answer]
Duplicate of #3379 (and #3981).

> Knowed work arounds:
> - `nodbus`/`dbus-user none` or `net none`, both don't work for all programs
> -  `env GTK_IM_MODULE=xim` and `env QT_IM_MODULE=xim`

> Problematic strace line (strace -f):
`connect(14, {sa_family=AF_UNIX, sun_path=@"/home/ubuntu01/.cache/ibus/dbus-AyGV49uw"}, 41) = -1 ECONNREFUSED (Connection refused)`
>
> Looks like ibus is rejecting the connection. Maybe because of the PID-namespace?

**Reply by Alex-Farol** [reply]
Nodbus fixed the problem.
